### PR TITLE
refactor: remove deprecated openai-error-handler shim and use error-handler

### LINKED ISSUE #766: [ENHANCEMENT] Remove deprecated openai-error-handler shim
### Problem (one or two sentences)

`openai-error-handler` remains as a deprecated compatibility shim that wraps `error-handler`, which has been the canonical source since [RooCodeInc/Roo-Code#10204](https://github.com/RooCodeInc/Roo-Code/pull/10204). Keeping this shim creates duplicate import paths and duplicate tests for the same `handleOpenAIError` behavior, increasing maintenance overhead.

### Context (who is affected and when)

Provider and embedder code that uses OpenAI-compatible error handling currently has references to the deprecated `openai-error-handler` shim, while the actual canonical implementation lives in `error-handler`.

This can make it unclear which module should be used when updating or reviewing error handling logic. It also leaves deprecated code and duplicate tests in the codebase even though the canonical implementation already exists.

### Desired behavior (conceptual, not technical)

OpenAI-compatible error handling should be managed through a single canonical utility: `error-handler`.

The deprecated `openai-error-handler` shim and its duplicate test file should be removed, and all related imports should reference `error-handler` directly. The existing `handleOpenAIError` behavior should remain unchanged.

### Constraints / preferences (optional)

_No response_

### Request checklist

- [x] I've searched existing Issues and Discussions for duplicates
- [x] This describes a specific problem with clear context and impact

### Zoo Code Task Links (optional)

_No response_

### Acceptance criteria (optional)

_No response_

### Proposed approach (optional)

_No response_

### Trade-offs / risks (optional)

_No response_

### PR DESCRIPTION
**Pull request by daewoongoh**
### Related GitHub Issue

Closes: #766

### Description

This PR removes the deprecated `openai-error-handler` compatibility shim and consolidates all related imports to use the canonical `error-handler` utility directly.

`error-handler` has been the canonical source for OpenAI-compatible error handling since [RooCodeInc/Roo-Code#10204](https://github.com/RooCodeInc/Roo-Code/pull/10204). The deprecated `openai-error-handler` module only acted as a shim, so keeping it added unnecessary duplication and maintenance overhead.

Key changes:

* Replaced all imports of the deprecated `openai-error-handler` shim with `error-handler`.
* Removed `src/api/providers/utils/openai-error-handler.ts`.
* Removed the duplicate `openai-error-handler.spec.ts` test file.
* Kept the actual OpenAI-compatible error handling behavior centralized in `error-handler.ts`.
* Continued relying on `error-handler.spec.ts`, which already covers the same behavior previously tested by the removed duplicate test.

This PR is intended to be a no-behavior-change refactor. It simplifies the provider utility structure while preserving the existing `handleOpenAIError` behavior.

### Test Procedure

```bash
pnpm check-types
pnpm test
```

### Pre-Submission Checklist

* [x] **Issue Linked**: This PR is linked to an approved GitHub Issue (see "Related GitHub Issue" above).
* [x] **Scope**: My changes are focused on the linked issue (one major feature/fix per PR).
* [x] **Self-Review**: I have performed a thorough self-review of my code.
* [x] **Testing**: New and/or updated tests have been added to cover my changes (if applicable).
* [x] **Documentation Impact**: I have considered if my changes require documentation updates (see "Documentation Updates" section below).
* [x] **Contribution Guidelines**: I have read and agree to the [Contributor Guidelines](/CONTRIBUTING.md).

### Screenshots / Videos

N/A

### Documentation Updates

* [x] No documentation updates are required.

### Get in Touch

hehegwk_23849

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved consistency of error messages across multiple AI provider and code index embedder flows by routing failures through a shared error handler (including proper provider-prefixed formatting and status preservation).
  * Removed reliance on the deprecated OpenAI-specific error handler so behavior is normalized across OpenAI-compatible services.
* **Tests**
  * Added a DeepSeek streaming failure test to verify the wrapped error message prefix and preserved status.
* **Chores**
  * Removed the obsolete OpenAI error handler unit test coverage.

### CODE REVIEW

**Comment by daewoongoh** [comment]
@coderabbitai review

File: src/api/providers/deepseek.ts [hunk 1]
```diff
@@ -137,7 +137,7 @@ export class DeepSeekHandler extends OpenAiHandler {
 				isAzureAiInference ? { path: OPENAI_AZURE_AI_INFERENCE_PATH } : {},
 			)
 		} catch (error) {
-			const { handleOpenAIError } = await import("./utils/openai-error-handler")
+			const { handleOpenAIError } = await import("./utils/error-handler")
```
Comment (edelauna) [inline src/api/providers/deepseek.ts]: Every other provider uses a static import for `handleOpenAIError` at the top of the file — is there a reason this one uses a dynamic import inside the catch block? If not, a static import would keep it consistent.

```suggestion
import { handleOpenAIError } from "./utils/error-handler"
```

(And the catch body becomes just `throw handleOpenAIError(error, "DeepSeek")`)

File: src/api/providers/__tests__/deepseek.spec.ts [hunk 2]
```diff
@@ -426,6 +426,14 @@ describe("DeepSeekHandler", () => {
 			expect(chunks).toContainEqual({ type: "reasoning", text: "router-style thought" })
 		})
 
+		it("should throw a wrapped error when the API call fails", async () => {
+			const apiError = Object.assign(new Error("Invalid API key"), { status: 401 })
+			mockCreate.mockRejectedValueOnce(apiError)
+
+			const stream = handler.createMessage(systemPrompt, messages)
+			await expect(stream.next()).rejects.toThrow("DeepSeek completion error: Invalid API key")
```
Comment (edelauna) [inline src/api/providers/__tests__/deepseek.spec.ts]: The test checks the message format, but `.status` is what the retry/backoff logic reads downstream — could we assert that it is preserved too?

```suggestion
			const err = await stream.next().catch((e) => e)
			expect(err.message).toBe("DeepSeek completion error: Invalid API key")
			expect((err as any).status).toBe(401)
```

**Review by edelauna — CHANGES_REQUESTED** [review]
Looks good - had some comments around test accuracy and import consistency.

File: src/api/providers/deepseek.ts (hunk 1 above)
Comment (daewoongoh) [reply src/api/providers/deepseek.ts]: There wasn’t a specific reason to keep this as a dynamic import.
I’ve updated DeepSeek to use the same static handleOpenAIError import as the other providers and simplified the catch block to call throw handleOpenAIError(error, "DeepSeek") directly.

File: src/api/providers/__tests__/deepseek.spec.ts (hunk 2 above)
Comment (daewoongoh) [reply src/api/providers/__tests__/deepseek.spec.ts]: I’ve updated the DeepSeek error test to capture the thrown error and assert both the wrapped message and the preserved status value.
This should cover the downstream retry/backoff logic that reads .status.

**Review by edelauna — APPROVED** [review]
Thanks for your contribution.
